cache.go

```diff
@@ -14,9 +14,14 @@ func (d *fronted) initCaching(cacheFile string) {
 
 func (d *fronted) prepopulateFronts(cacheFile string) {
 	bytes, err := os.ReadFile(cacheFile)
+	if os.IsNotExist(err) {
+		if err = os.MkdirAll(filepath.Dir(cacheFile), 0755); err != nil {
+			log.Errorf("Error creating cache directory: %v", err)
+			return
+		}
+	}
 	if err != nil {
-		// This is not a big deal since we'll just fill the cache later
-		log.Debugf("ignorable error: Unable to read cache file for prepopulation: %v", err)
```

Comment:
I don't think this is ignorable. By this point if it can't read the file it won't be able to write to it later.

Reply:
This isn't really true, as it just means no cache file has been created yet. That will be true every time on the first run, for example.

Reply:
> By this point if it can't read the file it won't be able to write to it later.
Sorry, by this I meant after including the change to create the parent directory.

However, even before this, while unlikely, it could still fail if it doesn't have read/write permissions on the directory.